### PR TITLE
Add PCRS 5cm imagery WMTS source for metropolitan France

### PR DESCRIPTION
**Pull request by flacombe**
Dar maintainers,

I propose to add a new 5cm imagery source for metropolitan France

GetCapabilities is here to check proposed layer: https://wxs.ign.fr/ortho/geoportail/wmts?SERVICE=WMTS&VERSION=1.0.0&REQUEST=GetCapabilities

Best regards

### CODE REVIEW

File: sources/europe/fr/PCRS.geojson [hunk 1]
```diff
@@ -0,0 +1,2289 @@
...
+            "text": "PCRS - République Française",
+            "required": true
+        },
+        "icon": "https://pcrs.beta.gouv.fr/images/icons/map.png",
+        "max_zoom": 21,
+        "min_zoom": 3,
+        "available_projections": [
+            "EPSG:2154"
+        ],
+        "license_url": "https://www.etalab.gouv.fr/licence-ouverte-open-licence/",
```
Comment (rbuffat) [inline sources/europe/fr/PCRS.geojson]: The `license_url` should point to a website that specifies the license of the data, not the license itself. E.g. ideally on the site linked here, it should say dataset PCRS rasters has license X.

File: sources/europe/fr/PCRS.geojson [hunk 2]
```diff
@@ -0,0 +1,2289 @@
...
+    "type": "Feature",
+    "properties": {
+        "id": "fr.pcrs",
+        "name": "PCRS raster",
+        "type": "tms",
+        "url": "https://wxs.ign.fr/ortho/geoportail/wmts?SERVICE=WMTS&REQUEST=GetTile&VERSION=1.0.0&LAYER=PCRS.LAMB93&STYLE=normal&FORMAT=image/jpeg&tileMatrixSet=LAMB93_5cm_EPSG&tileMatrix={zoom}&tileRow={y}&tileCol={x}",
+        "best": true,
+        "country_code": "FR",
+        "attribution": {
+            "url": null,
```
Comment (rbuffat) [inline sources/europe/fr/PCRS.geojson]: If there is no url it is better to leave the `url` out.

**Comment by rbuffat** [comment]
@flacombe Thanks for your PR!  Would it be possible for you to answer the questions I asked above?

File: sources/europe/fr/PCRS.geojson (hunk 1 above)
Comment (flacombe) [reply sources/europe/fr/PCRS.geojson]: Indeed, I only know https://geoservices.ign.fr/pcrs which isn't clear regarding the license.

Be sure I'll ask to complete it with appropriate mention next week.
All imagery from IGN/public services is Etalab/Open Licence v2 actually.

**Comment by flacombe** [comment]
Hi @rbuffat, it's all done now, thank you for interest

**Comment by flacombe** [comment]
Hello

I don't understand the warning provided with the check log.
I don't use any LinearRing geometry... where is it?

**Comment by rbuffat** [comment]
@flacombe I will try to have a look at it during the next few days!

File: sources/europe/fr/PCRS.geojson [hunk 3]
```diff
@@ -0,0 +1,2288 @@
...
+        "available_projections": [
+            "EPSG:2154"
+        ],
+        "license_url": "https://geoservices.ign.fr/pcrs",
+        "category": "photo"
+    },
+    "geometry": {
+        "type": "Polygon",
+        "coordinates": [
+            [
```
Comment (rbuffat) [inline sources/europe/fr/PCRS.geojson:24]: The issue here is that there are 4 levels of `[  [  [  [ `, but only `[ [ [ ` levels are needed for a Polygon. A Polygon is enclosed with a LinearRing. The library we use to parse the geometry was not able to parse it and thus emitted this error message mentioning LinearRings.

File: sources/europe/fr/PCRS.geojson [hunk 4]
```diff
@@ -0,0 +1,2288 @@
+{
+    "type": "Feature",
+    "properties": {
+        "id": "fr.pcrs",
+        "name": "PCRS raster",
+        "type": "tms",
+        "url": "https://wxs.ign.fr/ortho/geoportail/wmts?SERVICE=WMTS&REQUEST=GetTile&VERSION=1.0.0&LAYER=PCRS.LAMB93&STYLE=normal&FORMAT=image/jpeg&tileMatrixSet=LAMB93_5cm_EPSG&tileMatrix={zoom}&tileRow={y}&tileCol={x}",
```
Comment (rbuffat) [inline sources/europe/fr/PCRS.geojson]: While normally it is preferred to use a WMTS server, the projection of the TileMatrix is (as far as I know) not supported by all OSM editors. The WMS server mentioned here https://geoservices.ign.fr/services-web-experts-ortho seems to support EPSG:4326, which is widely supported: 
```
"type": "wms",
"url": "https://wxs.ign.fr/ortho/geoportail/r/wms?LAYERS=PCRS+LAMB93&STYLES=&CRS={proj}&BBOX={bbox}&FORMAT=image/jpeg&WIDTH={width}&HEIGHT={height}&VERSION=1.3.0&SERVICE=WMS&REQUEST=GetMap",
"available_projections": ["EPSG:3857","EPSG:4326"]
```

**Comment by rbuffat** [comment]
@flacombe I'm very sorry, it took a lot longer for me to look at it as I originally planned.

Using google I found the following link regarding the license situation: https://docs.pcrs.beta.gouv.fr/usages/contribution-openstreetmap
It would be much appreciated if somebody more familiar with the situation in France could have a look at it. Maybe  @grischard or @don-vip ?

I have to say I'm a bit confused. From the name "Plan Corps de Rue Simplifié" I would assume that the service would contain vector data, but it looks to be aerial images.

**Comment by flacombe** [comment]
Hi @rbuffat 

> Using google I found the following link regarding the license situation: https://docs.pcrs.beta.gouv.fr/usages/contribution-openstreetmap

I'm the author of this web page :)
The licence is granted by the French opendata general lawful framework
https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000044403448

As stated by
https://ghost.pcrs.beta.gouv.fr/content/files/2023/01/AGD-Saisine-ANCT-PCRS-20230124.pdf

> I have to say I'm a bit confused. From the name "Plan Corps de Rue Simplifié" I would assume that the service would contain vector data, but it looks to be aerial images.

PCRS is divided into two main data formats: raster and vector.
The raster part can be used as background aerial imagery suitable for OSM and the vector could be directly imported or used in quality control at least, when available (still no opendata currently).
https://docs.pcrs.beta.gouv.fr/contexte-reglementaire/geostandards#vecteur-et-raster

Let me know if those further resources can be suitable for license grant.

Best regards

File: sources/europe/fr/PCRS.geojson (hunk 4 above)
Comment (flacombe) [reply sources/europe/fr/PCRS.geojson]: Sounds good thank you

File: sources/europe/fr/PCRS.geojson (hunk 3 above)
Comment (flacombe) [reply sources/europe/fr/PCRS.geojson:24]: Sounds good thank you

**Comment by Marc-marc-marc** [comment]
@flacombe adding privacy_policy_url could be usefull, I also have splitted change for Volta map style in another PR to make this RP more readable

**Pull request merged**